contributors/design-proposals/security-model.md

```diff
@@ -0,0 +1,190 @@
+
+# Workloads
+
+The other face of Kubernetes exposed to customers is the workload environment.
+This section focuses on protecting workloads from the environment, and the
+environment from workloads.
+
+## Scheduled isolation
+
+Many users have use cases for scheduled isolation. This is where Kubernetes
+separates sensitive workloads from each other.
+
+1. A sensitive application might not tolerate other workloads on the same node.
+2. A user may bring a dedicated node.
+3. An application may need to run privileged containers.
+
+Both affinity and taints can achieve this today, but neither is protected. Both
+affinities and tolerations are specific on the "PodSpec", so anyone with "create
+pod" permissions can circumvent the mechanism.
+
+### Dedicated nodes
+
+Dedicated nodes require scheduling protection, because A's safety shouldn't be
+dependent on B's goodwill. Taint ACLs would suffer from the same problem as
+label ACLs. However, the "ownership" model can be applied successfully.
+
+For instance, a special "allowed-namespaces" taint would forbid pods outside of
+authorized namespaces. The pods would still be responsible for selecting the
+nodes.
+
+Alternatively, membership in a group, or a role, could be used to permit or
+forbid execution on a node. This might make more sense than forcing namespaces
+onto nodes, but probably requires the intentional ownership model.
+
+### Privileged pods
+
+A PodSecurityPolicy can prevent users from creating privileged containers.
+However, some use cases may be that privileged pods can only be created on
+dedicated nodes.
+
+For instance, a game server may want to run with host networking. These will
+need to be created with a privileged PSP. But many workloads may not tolerate
+sharing a host with privileged user pods.
+
+1. Privileged containers could taint a node.
+2. A cluster-level policy could restrict privileged containers to dedicated
+   nodes.
+3. A "NodeSecurityPolicy" could be introduced, which could forbid privileged
+   user containers.
+
+## Identity and Secrets
+
+Kubernetes is not intended to be a key management system. Customers should run a
+key management system (KMS) that emphasizes security (e.g., pinned memory, encrypted
+at rest, etc.). The chosen KMS can deliver appropriate secrets to containers.
+
+Bootstrapping identity may still fall to Kubernetes. Either containers are
+created with identifying resources (e.g., a cert), or the bootstrap protocol can
+physically verify the origin of a request (e.g., by inspecting OS resources).
+
+### Separation of privileges
+
+Imagine a pod with the following members and permissions:
+
+Permissions: logging (W), metrics (W), database (R/W)
+Containers: logger, metric agent, web service
+
+The pod's privileges can be further separated, giving each container only the
+permissions needed to function. This suggests that a KMS needs to be capable of
+distinguishing individual containers.
```

Comment:
I think we've always intended to support container scope privileges. If we want to bootstrap a KMS with k8s secrets, then it's pretty straight forward.

@mikedanese 
@cjcullen 

https://kubernetes.io/docs/concepts/configuration/secret/#use-case-secret-visible-to-one-container-in-a-pod